doc/nrf/libraries/modem/at_cmd_hook.rst

```diff
@@ -0,0 +1,98 @@
+.. _at_cmd_hook_readme:
+
+AT command hook library
+#######################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The AT command hook library allows any part of the application to register hooks to the AT API of the Modem library, to get a callback when matching AT commands are sent.
+
+Configuration
+*************
+
+You can enable the AT command hook library by setting the :kconfig:option:`CONFIG_AT_CMD_HOOK` Kconfig option.
+
+Usage
+*****
+
+To add an AT command hook, you must add the AT command filter and callback with the :c:macro:`AT_CMD_HOOK` macro in the :file:`at_cmd_hook.h` file.
+If the filter matches the first symbols of the AT command that is sent to the :c:func:`nrf_modem_at_cmd`, :c:func:`nrf_modem_at_printf` or :c:func:`nrf_modem_at_scanf` functions, the application is notified through the pre and post modem callbacks.
+
+.. note::
+   The AT command hooks does not trigger on calls to the :c:func:`nrf_modem_at_cmd_async` function.
+
+.. warning::
```

Comment:
```suggestion
.. note::
```
Lets keep this as notes and move this after line no: 32 because this content doesn't fall under the warning admonition.